polygeist/tools/cgeist/Test/Verification/sycl/targets.cpp

```diff
@@ -0,0 +1,21 @@
+// RUN: not clang++ -O0 %s -w -o - -S -emit-llvm -fsycl -fsycl-device-only \
+// RUN: -fsycl-targets=nvptx-unknown-unknown-syclmlir -nocudalib 2>&1 \
+// RUN: | FileCheck %s --check-prefix=CHECK-NVPTX
+
+// RUN: clang++ -O0 %s -o - -S -emit-llvm -fsycl -fsycl-device-only \
+// RUN: -fsycl-targets=spir-unknown-unknown-syclmlir 2>&1 \
+// RUN: | FileCheck %s --check-prefix=CHECK-32SPIR
+
+// RUN: clang++ -O0 %s -w -o - -S -emit-llvm -fsycl -fsycl-device-only \
+// RUN: -fsycl-targets=spir64-unknown-unknown-syclmlir 2>&1 \
+// RUN: | FileCheck %s --check-prefix=CHECK-64SPIR
+
+#include <sycl/sycl.hpp>
+
+// CHECK-NVPTX: Cannot lower SYCL target "nvptx-unknown-unknown-syclmlir" to LLVM
+
+// CHECK-32SPIR: Using the 32-bits spir target may lead to errors when lowering the `sycl` dialect.
```

Comment:
Makes me wonder whether we should even support this at all.

Reply:
Just using `index` instead of `i64` in some pieces of the code should fix this